src/main/java/org/threadly/litesockets/NoThreadSocketExecuter.java

```diff
@@ -46,18 +53,27 @@ public NoThreadSocketExecuter() {
    */
   public void wakeup() {
     if(commonSelector != null && commonSelector.isOpen()) {
+      wakeUp = true;
       commonSelector.wakeup();
     }
   }
 
   @Override
   public void setClientOperations(final Client client) {
     ArgumentVerifier.assertNotNull(client, "Client");
-    if(!clients.containsKey(client.getChannel()) || client.isClosed()) {
+    if(!clients.containsKey(client.getChannel())) {
       clients.remove(client.getChannel());
       return;
     }
-    if(client.getChannel().isConnectionPending()) {
+    if(client.isClosed()) {
+      clients.remove(client.getChannel());
+      ListenableFuture<?> lf = schedulerPool.submit(new RemoveFromSelector(commonSelector, client));
+      lf.addListener(new Runnable() {
+        @Override
+        public void run() {
+          PortUtils.closeQuietly(client.getChannel());
```

Comment:
well, we dont want to do it till both selectors are removed from for 1, and removing a client might not mean we want to close it.  in this code it does, but that will probably change since addToSelector was techincally doing removes also.